src/v/cluster/rm_stm.cc

```diff
@@ -1774,19 +1775,32 @@ ss::future<> rm_stm::do_try_abort_old_tx(model::producer_identity pid) {
 
     _mem_state.expected.erase(pid);
 
-    std::optional<prepare_marker> marker;
-    auto prepare_it = _mem_state.preparing.find(pid);
-    if (prepare_it != _mem_state.preparing.end()) {
-        marker = prepare_it->second;
+    std::optional<model::tx_seq> tx_seq;
+    if (!is_transaction_ga()) {
+        auto prepare_it = _mem_state.preparing.find(pid);
+        if (prepare_it != _mem_state.preparing.end()) {
+            tx_seq = prepare_it->second.tx_seq;
+        }
+        prepare_it = _log_state.prepared.find(pid);
+        if (prepare_it != _log_state.prepared.end()) {
+            tx_seq = prepare_it->second.tx_seq;
+        }
     }
-    prepare_it = _log_state.prepared.find(pid);
-    if (prepare_it != _log_state.prepared.end()) {
-        marker = prepare_it->second;
+
+    if (!tx_seq) {
+        tx_seq = get_tx_seq(pid);
     }
 
-    if (marker) {
+    if (is_transaction_ga() && !tx_seq) {
+        vlog(
+          _ctx_log.error,
+          "Can not find tx_seq for pid({}) to expire old tx",
+          pid);
+    }
+
+    if (tx_seq) {
         auto r = co_await _tx_gateway_frontend.local().try_abort(
-          marker->tm_partition, marker->pid, marker->tx_seq, _sync_timeout);
+          model::partition_id(0), pid, *tx_seq, _sync_timeout);
```

Comment:
i think this change, where the partition is now fixed to partition `0`, deserves a sentence or two in the commit message. the reason is that it feel like a structural change that goes beyond what is described in the commit message. it is probably obvious to folks steeped in transaction stuff, but it should still be consumable by others.

Reply:
added a comment